R/topics.R

```diff
@@ -167,3 +167,59 @@ topic_must <- function(message, topic) {
     x = paste0("Not ", encodeString(topic, quote = "'"))
   ))
 }
+
+content_info <- function(content_entry, index, pkg, section) {
+
+  if (!grepl(".*::.*", content_entry)) {
+    topics <- pkg$topics[select_topics(content_entry, pkg$topics),]
+    tibble::tibble(
+      path = topics$file_out,
+      aliases = purrr::map2(
+        topics$funs,
+        topics$name,
+        ~ if (length(.x) > 0) .x else .y
+      ),
+      name = list(topics$name),
+      title = topics$title,
+      icon = find_icons(topics$alias, path(pkg$src_path, "icons"))
+    )
+  } else { # topic from another package
+    names <- strsplit(content_entry, "::")[[1]]
+    if (!rlang::is_installed(names[1])) {
+      abort(
+        sprintf(
+          "%s must be installed if it is mentioned in the reference index.",
+          names[1]
+        )
+      )
+    }
+
+    path <- downlit::href_topic(names[2], names[1])
+    if (is.na(path)) {
+      abort(
+        sprintf(
+          "Could not find an href for topic %s of package %s",
+          names[2], names[1]
+        )
+      )
+    }
+
+    rd_name <- fs::path_ext_set(fs::path_file(path), "Rd")
+    # adapted from printr
+    db <- tools::Rd_db(names[1])
+    Rd <- db[[rd_name]]
+    # adapted from printr
+    # https://github.com/yihui/printr/blob/0267c36f49e92bd99e5434f695f80b417d14e090/R/help.R#L69
+    sections <- sub('^\\\\', '', unlist(lapply(Rd, attr, 'Rd_tag')))
+    title <- as.character(Rd[sections == "title"][[1]])
```

Comment:
Pull this out into a separate named function just to clarify the purpose. 

@yihui can you please confirm that it's ok for us to use your GPL licensed code in our MIT licensed package?

Reply:
Yes.

Reply:
Thank you!

However I am now extremely embarrassed to have at last remembered where else code for parsing Rd files lives... in pkgdown itself! :woman_facepalming: :see_no_evil:

Reply:
Still using a bit of printr logic though, for finding the Rd.